src/msw/dc.cpp

```diff
@@ -817,7 +817,30 @@ void wxMSWDCImpl::DoCrossHair(wxCoord x, wxCoord y)
 
 void wxMSWDCImpl::DoDrawLine(wxCoord x1, wxCoord y1, wxCoord x2, wxCoord y2)
 {
-    wxDrawLine(GetHdc(), XLOG2DEV(x1), YLOG2DEV(y1), XLOG2DEV(x2), YLOG2DEV(y2));
+#if 1
+    // If we are going to draw a vertical or horizontal line with pen with solid colour and flat cap
+    if ( ( x1 == x2 || y1 == y2 ) &&
+         ( !m_pen.IsOk() || ( m_pen.GetColour().IsSolid() && ( m_pen.GetWidth() == 1 || m_pen.GetCap() == wxCAP_BUTT ) ) ) )
```

Comment:
I think this should be
```suggestion
         ( !m_pen.IsOk() || ( m_pen.GetStyle() == wxPENSTYLE_SOLID && ( m_pen.GetWidth() == 1 || m_pen.GetCap() == wxCAP_BUTT ) ) ) )
```

I'd also like to wrap this line to fit in 80 columns.